group-generators/helpers/data-providers/dune/provider.ts

```diff
@@ -0,0 +1,193 @@
+import { DuneErrorFactory } from './errors';
+import {
+  QueryParams,
+  ExecutionState,
+  ExecuteQuery,
+  ExecutionStatusComplete,
+  ExecutionResults,
+  Row,
+} from './types';
+import { FetchedData } from 'topics/group';
+
+//Dune requires an API key
+export class DuneProvider {
+  /**
+   * Use this method to query Dune Analytics for your data.
+   * @param queryId The Dune QueryID e.g. 2034748 https://dune.com/queries/2034748"
+   * @param duneEthAddressColumn e.g. “The ETH Address Winner”  for the Dune Query https://dune.com/queries/2034748 -> used to match the ETH addresses
+   * @param queryParams optional! e.g. {address: '0x3CAaE25Ee616f2C8E13C74dA0813402eae3F496b', blocknumber: '66257668', chain: 'arbitrum'} for the Dune Query https://dune.com/queries/1618116
+   * @returns The FetchedData object in the <FetchedData> Type
+   */
+  public async executeQuery(
+    queryId: number,
+    duneEthAddressColumn: string,
+    queryParamsObject?: Record<string, string> | null | undefined
+  ): Promise<FetchedData> {
+    const queryParams = queryParamsObject
+      ? (queryParamsObject as QueryParams)
+      : undefined;
+    const { execution_id, state } = await this.executeNewQuery(
+      queryId,
+      queryParams
+    );
+
+    console.log(
+      `\n dune_execution_id is ${execution_id} initial state is ${state} \n depending on size of query and Dune response time, this can take max 30 mins \n`
+    );
+
+    await this.getExecutionStatus(execution_id);
+    const results = await this.getResults(execution_id);
+    return this.formatData(results.result.rows, duneEthAddressColumn);
+  }
+
+  public async executeQueryCount(
```

Comment:
Same problem here, you need to create a type that contains all your arguments.

Reply:
Thanks @MartinGbz, I've updated the args